tg_bot/modules/log_channel.py

```diff
@@ -0,0 +1,263 @@
+                    func,
+                )
+
+            return result
+
+        return log_action
+
+    @run_async
+    def log_resource(bot: Bot, update: Update):
+        entities = update.effective_message.parse_entities()
+        caption_entities = update.effective_message.parse_caption_entities()
+        LOGGER.info(entities)
+        chat = update.effective_chat  # type: Optional[Chat]
+        log_chat = sql.get_chat_log_channel(chat.id)
+
+        url = None
+        tags = []
+
+        if log_chat:
+            for descriptor, entity in entities.items():
+                if descriptor["type"] == MessageEntity.HASHTAG:
+                    tags.append(entity)
+            if tags:
+                tags.sort()
+
+            log_entities_list(bot, chat, entities, log_chat, tags, update)
+
+            log_entities_list(bot, chat, caption_entities, log_chat, tags, update)
+
+
+    def log_entities_list(bot, chat, entities, log_chat, tags, update):
+        for descriptor, entity in entities.items():
+            result = (
+                f"<b>Risorsa inviata da @{update.effective_user.username}:</b>\n"
+            )
+
+            if descriptor["type"] == MessageEntity.URL:
+                try:
+                    response = requests.get(entity)
+                    if response.status_code == requests.codes.ok:
+                        result += f"{entity}"
+                        extracted = tldextract.extract(entity)
+                        if f"#{extracted.domain}" not in tags:
+                            tags.append(f"#{extracted.domain}")
+                            tags.sort()
+                        if tags:
+                            tags_string = " ".join(tags)
+                            result += f"\n\nTags:\n{tags_string}"
+                        send_log(bot, log_chat, chat.id, result)
+                except Exception as e:
+                    LOGGER.info(f"Resource {entity} is not a valid url")
+                    LOGGER.error(e)
+            elif descriptor["type"] == MessageEntity.TEXT_LINK:
+                try:
+                    response = requests.get(descriptor["url"])
+                    if response.status_code == requests.codes.ok:
+                        result += f'{descriptor["url"]}'
+                        extracted = tldextract.extract(descriptor["url"])
+                        if f"#{extracted.domain}" not in tags:
+                            tags.append(f"#{extracted.domain}")
+                            tags.sort()
+                        if tags:
+                            tags_string = " ".join(tags)
+                            result += f"\n\nTags:\n{tags_string}"
+                        send_log(bot, log_chat, chat.id, result)
+                except Exception as e:
+                    LOGGER.info(f"Resource {entity} is not a valid url")
+                    LOGGER.error(e)
+
```

Comment:
Ahhhh, da generalizzare la parte interna dell'if e sbatterla in un metodo. Effettivamente ho sempre molto sonno quando lavoro sul bot. Me ne occuperò quanto prima, ma non riuscirò prima di questo fine settimana. Se qualcuno riesce prima di me è ben accetto, altrimenti ci penso io sabato o domenica.